x-pack/legacy/plugins/monitoring/public/lib/setup_mode.js

```diff
@@ -70,11 +73,41 @@ const notifySetupModeDataChange = (oldData) => {
 
 export const updateSetupModeData = async (uuid, fetchWithoutClusterUuid = false) => {
   const oldData = setupModeState.data;
-  setupModeState.data = await fetchCollectionData(uuid, fetchWithoutClusterUuid);
-  if (get(setupModeState.data, '_meta.isOnCloud', false)) {
+  const data = await fetchCollectionData(uuid, fetchWithoutClusterUuid);
+  setupModeState.data = data;
+  if (get(data, '_meta.isOnCloud', false)) {
     return toggleSetupMode(false); // eslint-disable-line no-use-before-define
   }
   notifySetupModeDataChange(oldData);
+
+  const globalState = angularState.injector.get('globalState');
+  const clusterUuid = globalState.cluster_uuid;
+  if (!clusterUuid) {
+    const liveClusterUuid = get(data, '_meta.liveClusterUuid');
+    const migratedEsNodes = Object.values(get(data, 'elasticsearch.byUuid', {}))
+      .filter(node => node.isPartiallyMigrated || node.isFullyMigrated);
+    if (migratedEsNodes.length > 0) {
+      setNewlyDiscoveredClusterUuid(liveClusterUuid);
+    }
+  }
+};
+
+export const disableElasticsearchInternalCollection = async () => {
+  checkAngularState();
```

Comment:
I know this is from previous PRs, but is there any way to avoid keeping angular state? Maybe each function can take an `injector` as an argument? 

Or, maybe even use: `const $injector = await chrome.dangerouslyGetActiveInjector();` ?

Reply:
I didn't know about this! I'll see if I can find a way to use this instead. Thanks!

Reply:
Actually, now that I think about it, I think this is okay, since this process/migration probably won't be used that much (and we won't have to touch this code for a while)

Reply:
I guess maybe I should ask you to clarify about what you dislike about it.

> Maybe each function can take an injector as an argument?
This is hard because it means each caller will need to have access to angular bits. Right now, the way I tried to construct this was for the view layer (which is in angular) to control setting up these angular bits so other parts of the code didn't need to care.

> Or, maybe even use: const $injector = await chrome.dangerouslyGetActiveInjector();

I could use that, but I don't know the advantage. Do you think the need for views to handle this setup is unnecessary? Or is it something about the usage?

Reply:
I guess this is more of a personal preference, but each time you want to use a method from `setup_mode.js` you have to remember if you already set the angular state (or you still need to do that in one of the parents somewhere).

I like methods that don't require any setup/state and you can just use naively/blindly

Reply:
I agree with your thinking, but I don't know what a better way will be in this scenario. If we don't do a setup, then every consumer of any of the public functions exposed in `setup_mode` will need access to both a scope and injector, and it's possible a consumer is deeply nested in some React component and can't easily access them easily.

Maybe I'm missing something and I'm all ears for something better!

Reply:
@chrisronline What I meant is to implement something like:
```javascript
const getScope = () => angular.element('.application').scope();
const getInjector = () => angular.element('.application').injector();
```
inside your `setup_mode.js` and get rid of `initSetupModeState` and `checkAngularState`.

Up to you though, I'm fine with the way it is right now

Reply:
@ycombinator Any strong thoughts here either way?

Reply:
I don't have a strong opinion either way — sounds like there's pros/cons to either approach. To break the tie and move this forward, is there a certain way we're doing things right now, in code outside this PR? If so, I'd go with that pattern for now and maybe file a `discuss` issue to hash out if/how we want to change this across the board later.

Reply:
FWIW, I did a code search in kibana for usage of `angular.element` and didn't find much use which lends me to think that maybe there are some gotchas to that approach? I'm not aware of any offhand, but we're definitely not using `angular.element` anywhere else in monitoring

https://github.com/elastic/kibana/search?l=JavaScript&q=angular.element